ui/ducks/send/send.js

```diff
@@ -2115,6 +2191,38 @@ export function updateSendAsset(
           );
         }
       }
+      if (state.UNS) {
+        if (
+          state.UNS.resolution ===
+            state.send.draftTransactions[state.send.currentTransactionUUID]
+              .recipient.address &&
+          state.UNS.domainName
+        ) {
+          const object = await swapToken(state.UNS.domainName, asset);
+          if (object.error) {
+            if (
+              object.error === 'UnspecifiedCurrency' ||
+              object.error === 'RecordNotFound'
+            ) {
+              asset.error = UNS_CURRENCY_SPEC_ERROR;
+            } else if (object.error === 'UnsupportedCurrency') {
+              asset.error = UNS_CURRENCY_ERROR;
+            } else {
+              asset.error = UNS_UNKNOWN_ERROR;
+            }
+            await dispatch(actions.updateAsset({ asset, initialAssetSet }));
+            throw new Error('No address associated with this token');
+          } else {
+            dispatch(unsLookup(object));
+            await dispatch(
+              updateRecipient({
+                address: object.address,
+                nickname: object.unsName,
+              }),
+            );
+          }
+        }
+      }
```

Comment:
This code is duplicated can you help me understand what this portion of the code is doing?

Reply:
we had this twice because when you swap a token it can either be a native token or a multi chain token